sdk-tests/src/test/java/io/dapr/it/state/AbstractStateClientIT.java

```diff
@@ -249,6 +250,52 @@ public void saveUpdateAndGetStateWithEtag() {
     assertEquals("data in property B2", myDataResponse.getValue().getPropertyB());
   }
 
+  @Test
+  public void saveUpdateAndGetNullStateWithEtag() {
+    // The key use to store the state and be updated using etags
+    final String stateKey = UUID.randomUUID().toString();
+    DaprClient daprClient = buildDaprClient();
+    MyData data = new MyData();
+    data.setPropertyA("data in property A");
+    data.setPropertyB("data in property B");
+
+    // Get state to validate case for key not found.
+    State<MyData> stateNotFound = daprClient.getState(STATE_STORE_NAME, stateKey, MyData.class).block();
+    assertEquals(stateKey, stateNotFound.getKey());
+    assertNull(stateNotFound.getValue());
+    assertNull(stateNotFound.getEtag());
+    assertNull(stateNotFound.getOptions());
+    assertNull(stateNotFound.getError());
+    assertEquals(0, stateNotFound.getMetadata().size());
+
+    // Set non null value
+    daprClient.saveState(STATE_STORE_NAME, stateKey, null, data, null).block();
+
+    // Get state to validate case for key with value.
+    State<MyData> stateFound = daprClient.getState(STATE_STORE_NAME, stateKey, MyData.class).block();
+    assertEquals(stateKey, stateFound.getKey());
+    assertNotNull(stateFound.getValue());
+    assertEquals("data in property A", stateFound.getValue().getPropertyA());
+    assertEquals("data in property B", stateFound.getValue().getPropertyB());
+    assertNotNull(stateFound.getEtag());
+    assertFalse(stateFound.getEtag().isEmpty());
+    assertNull(stateFound.getOptions());
+    assertNull(stateFound.getError());
+    assertEquals(0, stateFound.getMetadata().size());
```

Comment:
in this case whenever a metadata field is returned it is always empty?

Reply:
It is possible. Today, Redis (the state store we use for ITs) does not do that.